components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -1684,7 +1684,20 @@ std::vector<base::WeakPtr<Tool>> ConversationHandler::GetTools() {
       associated_content_manager_->HasAssociatedContent(), GetCurrentModel());
   std::ranges::transform(conversation_tools, std::back_inserter(tools),
                          [](auto& tool) { return tool->GetWeakPtr(); });
-  // TODO(petemill): Concat with tools from other areas (content, browser, etc.)
+
+  // Add browser-level tools
+  auto browser_tools = ai_chat_service_->GetBrowserTools();
+
+  // Filter out memory tool for temporary conversations
+  if (GetIsTemporary()) {
+    std::erase_if(browser_tools, [](const auto& tool) {
+      return tool && tool->Name() == mojom::kMemoryStorageToolName;
+    });
+  }
```

Comment:
Perhaps we need something like:
```
virtual SupportsConversation(ConversationHandler* conversation) const
```
on `Tool` so that we can contain the logic in the Tool itself?

In the Content Agent branch, I have 
```
virtual bool SupportsConversationCapability(
      mojom::ConversationCapability conversation_capability) const;
```
But passing the whole `ConversationHandler` would be more flexible. If that's passing too much and hinders testability we could just add ad-hoc properties to `SupportsConversation` e.g.
```
virtual bool SupportsConversation(
      mojom::ConversationCapability conversation_capability, bool is_temporary) const;
```

Reply:
I ended with `SupportsConversationCapability(bool is_temporary)` due to better testability. https://github.com/brave/brave-core/pull/30660/commits/68834cb97edfc2c151af69ee10302409e044a706